flink-yarn/src/main/java/org/apache/flink/yarn/YarnClusterDescriptor.java

```diff
@@ -1241,7 +1241,6 @@ private void failSessionDuringDeployment(YarnClient yarnClient, YarnClientApplic
 			// call (we don't know if the application has been deployed when the error occured).
 			LOG.debug("Error while killing YARN application", e);
 		}
-		yarnClient.stop();
```

Comment:
I can see other caller of `failSessionDuringDeployment` such as L525 and L511. Why do you remove this line. IIUC now we don't stop yarnClient in these two cases.

Reply:
In L525 and L511, it will throw `YarnDeploymentException` and then yarn client will be stopped in `CliFrontend->runProgram->clusterDescriptor.close()`. We should not stop yarn client twice. 

We create yarn client outside and i think we should always use `clusterDescriptor.close()` to stop the yarn client.

Reply:
Thanks for your clarification. Sounds reasonable.